### PR TITLE
Improve "Getting Started" docs

### PR DESCRIPTION
**Pull request by mehaase**
_I messed up #62 because I made that PR from master. Here I'm making the same PR from a separate branch._

For #19, I notice there is already a PR to improve the API docs, but not for the "Getting Started" docs. I spent today getting started with wsproto, so I wrote the docs that I wish had existed when I started this morning. It expands the existing prose, adds a diagram, and adds some sample code for clients and servers. Let me know what you think!

### CODE REVIEW

File: docs/source/basic-usage.rst [hunk 1]
```diff
@@ -14,27 +15,118 @@ The main class you'll be working with is the
...
+        elif isinstance(event, ConnectionClosed):
+            print('Connection closed: code={} reason={}'.format(
+                event.code, event.reason))
+        elif isinstance(event, TextReceived):
+            print('Received message: {}'.format(event.data))
+        else:
+            print('Unknown event: {!r}'.format(event))
+
+For a more complete example, see this `toy websocket server
+<https://gist.github.com/mehaase/772efaf451eca7a5c3ce7e72ebaefe4e#file-wsproto_server_sync-py>`_.
```
Comment (Kriechi) [inline docs/source/basic-usage.rst]: Why not put that into a new `example/toy_server.py` file?

**Comment by mehaase** [comment]
The CI linter job failed, which is strange because this PR only changes one `.rst` file and doesn't touch any Python code:

```
$ tox -- --verbose
GLOB sdist-make: /home/travis/build/python-hyper/wsproto/setup.py
lint create: /home/travis/build/python-hyper/wsproto/.tox/lint
lint installdeps: prospector==0.12.7
lint inst: /home/travis/build/python-hyper/wsproto/.tox/dist/wsproto-0.11.0.zip
lint installed: astroid==2.0.4,dodgy==0.1.9,flake8==3.5.0,flake8-polyfill==1.0.2,h11==0.7.0,isort==4.3.4,lazy-object-proxy==1.3.1,mccabe==0.6.1,pep8-naming==0.7.0,prospector==0.12.7,pycodestyle==2.0.0,pydocstyle==2.1.1,pyflakes==2.0.0,pylint==2.1.1,pylint-celery==0.3,pylint-common==0.2.5,pylint-django==2.0.2,pylint-flask==0.5,pylint-plugin-utils==0.4,PyYAML==3.13,requirements-detector==0.6,setoptconf==0.2.0,six==1.11.0,snowballstemmer==1.2.1,typed-ast==1.1.0,wrapt==1.10.11,wsproto==0.11.0
lint runtests: PYTHONHASHSEED='1459343171'
lint runtests: commands[0] | prospector
Traceback (most recent call last):
  File "/home/travis/build/python-hyper/wsproto/.tox/lint/bin/prospector", line 11, in <module>
    sys.exit(main())
  File "/home/travis/build/python-hyper/wsproto/.tox/lint/lib/python3.6/site-packages/prospector/run.py", line 165, in main
    prospector.execute()
  File "/home/travis/build/python-hyper/wsproto/.tox/lint/lib/python3.6/site-packages/prospector/run.py", line 53, in execute
    for tool in self.config.get_tools(found_files):
  File "/home/travis/build/python-hyper/wsproto/.tox/lint/lib/python3.6/site-packages/prospector/config/__init__.py", line 45, in get_tools
    config_result = tool.configure(self, found_files)
  File "/home/travis/build/python-hyper/wsproto/.tox/lint/lib/python3.6/site-packages/prospector/tools/pylint/__init__.py", line 172, in configure
    linter.reset_options()
  File "/home/travis/build/python-hyper/wsproto/.tox/lint/lib/python3.6/site-packages/prospector/tools/pylint/linter.py", line 30, in reset_options
    OptionsManagerMixIn.__init__(self, usage=PyLinter.__doc__, quiet=True)
TypeError: __init__() got an unexpected keyword argument 'quiet'
ERROR: InvocationError for command '/home/travis/build/python-hyper/wsproto/.tox/lint/bin/prospector' (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   lint: commands failed
The command "tox -- --verbose" exited with 1.
```

I think there's some mismatch between the version of prospector and the version of pylint being used. I will open up a separate PR for that.

**Comment by Kriechi** [comment]
Overall this looks good! 👍 
I would prefer to move your gists into an `example/` directory, so people who checkout the repo get all the information they are looking for.

**Comment by mehaase** [comment]
Ok, I've rebased, added some more content, and moved the example code from gists into `example/`.

**Comment by Kriechi** [comment]
This looks great! 👍 🎉 
Only our linter is not happy yet: https://travis-ci.org/python-hyper/wsproto/jobs/425746317#L487
Would you mind fixing this - then we can merge!

**Comment by Kriechi** [comment]
Thanks! 👍 🎉

**Pull request merged**